sorc/chgres_cube.fd/wam_climo_data.f90

```diff
@@ -0,0 +1,2758 @@
+                day,df,dfa,apd,apdf,apt(4),xlong
+      integer :: isw
+      integer :: iyr
+      integer :: mp,ii,jg,lt,ierr,ifun,n,j
+!x    character*4:: isd(3),ist(2),nam(2) 
+
+   end module gettemp_mod
+
+! ----------------------------------------------------------------------
+!> Entry routine to get WAM needed temperature and composition profiles.
+!!
+!! Calculate temperature at each grid point useing nrlmsise00_sub
+!! @param [in] iday(ngay) calendat date 
+!! @param [in] ngay dimension length of iday
+!! @param [in] xlat(nlat) latitudes
+!! @param [in] nlat dimension length of xlat
+!! @param [in] pr(np) pressure in vertical
+!! @param [in] np dimension length of pr
+!! @param [out] temp temperature
+!! @param [out] n_o single oxygen number
+!! @param [out] n_o2 oxygen number
+!! @param [out] n_n2 nitrogen number
+!!
+!! @author Hann-Ming Henry Juang  NCEP/EMC
+      subroutine gettemp(iday,nday,xlat,nlat,pr,np,temp,n_o,n_o2,n_n2)
+      implicit none
+      integer, intent(in) :: nday                 ! number of days
+      integer, intent(in) :: nlat                 ! number of latitudes
+      integer, intent(in) :: np                   ! number of pressure layer
+      real,    intent(in) :: pr(np)               ! pressure in mb
+      real,    intent(in) :: xlat(nlat)           ! latitude in degree
+      integer, intent(in) :: iday(nday)           ! calender day
+      real,   intent(out) :: temp(np,nlat,nday)   ! temperature
+      real,   intent(out) :: n_o(np,nlat,nday)    ! number density of o
+      real,   intent(out) :: n_o2(np,nlat,nday)   ! number density of o2
+      real,   intent(out) :: n_n2(np,nlat,nday)   ! number density of n2
+      real                :: alt(np,nlat,nday)    ! altitude in km
+      real                :: d(9),t(2),sw(25),ap(7),ut,xlong,xlst,f107, &
+                             f107a
+      integer             :: k,il,ip
+! set magnetic index average value
+      data ap/7*9./
+! set swich 7,8,10,14  zero to avoid diurnal changes in output temperatu
+! swich #7 is for diurnal,#8 is for semidiurnal,# 10 is for all ut/longi
+! effect,#14 is for terdiurnal
+      data sw/1.,1.,1.,1.,1.,1.,0.,0.,1.,0.,1.,1.,1.,0.,1.,1.,1.,1.,1., &
+      1.,1.,1.,1.,1.,1./
+! set 10.7cm flux be average value
+      f107=150.
+      f107a=150.
+! turn on swich
+      call tselec(sw)
+! set longitude, ut, local time , it should not make difference to outpu
+      ut=0.
+      xlong=0.
+      xlst=ut/3600.+xlong/15.
+! calculate temperature for each lat,pres level,day
+      do k=1,nday
+      do il=1,nlat
+      do ip=1,np
+      call ghp7(iday(k),ut,alt(ip,il,k),xlat(il),xlong,xlst,f107a,f107, &
+      ap,d,t,pr(ip))
+      temp(ip,il,k)=t(2)
+      n_o(ip,il,k)=d(2)
+      n_o2(ip,il,k)=d(4)
+      n_n2(ip,il,k)=d(3)
+      enddo
+      enddo
+      enddo
+      end subroutine gettemp
+
+!-----------------------------------------------------------------------
+!> The nrlmsise-00 subroutine gtd7.
```

Comment:
Not sure what this line is saying, looks like it should be removed. The first sentence of documentation for a subprogram should be a short description of the subprogram. It will be used as the "brief" description in the documentation, which appears a lot. So make the first sentence of the documentation block legible to users.